ENCODE_replaced_cleaner.py

```diff
@@ -0,0 +1,115 @@
+        description=__doc__, epilog=EPILOG,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    parser.add_argument('--key',
+                        default='default',
+                        help="The keypair identifier from the keyfile.  \
+                        Default is --key=default")
+    parser.add_argument('--keyfile',
+                        default=os.path.expanduser("~/keypairs.json"),
+                        help="The keypair file.  Default is --keyfile=%s" % (os.path.expanduser("~/keypairs.json")))
+    parser.add_argument('--debug',
+                        default=False,
+                        action='store_true',
+                        help="Print debug messages.  Default is False.")
+    args = parser.parse_args()
+    return args
+
+
+def retreive_list_of_replaced(object_to_inspect_acc,
+                              connection):
+    to_return_list = [object_to_inspect_acc]
+    objects_to_inspect = encodedcc.get_ENCODE('search/?type=Item&accession=' +
+                                              object_to_inspect_acc,
+                                              connection)['@graph']
+    if objects_to_inspect:
+        for object_to_inspect in objects_to_inspect:
+            if object_to_inspect.get('alternate_accessions'):
+                for acc in object_to_inspect.get('alternate_accessions'):
+                    to_return_list.extend(
+                        retreive_list_of_replaced(acc, connection))
+                return to_return_list
+            else:
+                return to_return_list
+    else:
+        return to_return_list
+
+
+def main():
+    args = getArgs()
+    key = encodedcc.ENC_Key(args.keyfile, args.key)
+    connection = encodedcc.ENC_Connection(key)
+    profiles = encodedcc.get_ENCODE("/profiles/", connection)
+    for object_type in profiles.keys():
+        uuid_2_alternate_accessions = {}
+        alternate_accessions_sets = []
+        objects = encodedcc.get_ENCODE('search/?type=' + object_type,
+                                       connection)['@graph']
+        for entry in objects:
+            if entry.get('alternate_accessions'):
+                replaced_objects_accessions = []
+                for acc in entry.get('alternate_accessions'):
+                    replaced_objects_accessions.extend(
+                        retreive_list_of_replaced(acc,
+                                                  connection))
+                if sorted(list(set(replaced_objects_accessions))) != sorted(
+                   entry.get('alternate_accessions')):
+                    uuid_2_alternate_accessions[entry['uuid']] = \
+                        set(replaced_objects_accessions)
+                    alternate_accessions_sets.append(
+                        set(replaced_objects_accessions))
+        for uuid in uuid_2_alternate_accessions.keys():
+            uuid_sets_counter = 0
+            for entry in alternate_accessions_sets:
```

Comment:
If we have a situation in which A replaces B, B replaces C, C replaces D. The uuid_2_alternate accessions will have A replaces (B,C,D) and B replaces (C,D) the condition will eliminate patch of B, since its set (C,D) is found as subset of (B,C,D).